lib/modules/csv_map_parser.rb

```diff
@@ -1,6 +1,32 @@
 require 'csv'
 
 module CsvMapParser
+  def self.ch2_map2_categorical
```

Comment:
Generally this approach works fine, but as there are many different constants used within the same function (`categ_country_stats`), I would have tried to use some metaprogramming.

An idea might be to define a constant hash like the following:

STATS = {
  method_name1: filename1,
  method_name2: filename2,
  ...
}

Then you could use the `define_method` function like:

```
STATS.each do |method_name, filename|
  define_method(method_name) do
    categ_country_stats(filename)
  end
end
```

The above might not be syntactically correct but that's the overall idea.

Reply:
You should be able to use `extend self` within the module. This basically allows to call methods without `self`